tfutils/model.py

```diff
@@ -1,254 +1,216 @@
+             activation='relu',
+             weight_decay=None,
+             in_layer=None):
+        if in_layer is None: in_layer = self.output
+        if weight_decay is None: weight_decay = 0.
         in_shape = in_layer.get_shape().as_list()[-1]
-        self.conv_counter += 1
-        self.global_counter += 1
-        if name is None:
-            name = 'conv' + str(self.conv_counter)
-        with tf.name_scope(name) as scope:
-            kernel = tf.Variable(
-                tf.truncated_normal([ksize, ksize, in_shape, out_shape],
-                                     dtype=tf.float32, stddev=stddev,
-                                     seed=self.seed),
-                name=name+'_weights')
-            conv = tf.nn.conv2d(in_layer, 
-                                kernel, 
-                                [1, stride, stride, 1],
-                                padding=padding,
-                                name=name)
-            biases = tf.Variable(tf.constant(bias,
-                                             shape=[out_shape],
-                                             dtype=tf.float32), 
-                                 name=name+'_biases')
-            conv_bias = tf.nn.bias_add(conv, biases)
-            conv_out = tf.nn.relu(conv_bias, name=scope)
-            self.parameters += [kernel, biases]
-            self.architecture[name] = {'input': in_layer.name,
-                                       'type': 'conv',
-                                       'num_filters': out_shape,
-                                       'stride': stride,
-                                       'kernel_size': ksize,
-                                       'padding': padding, 
-                                       'bias': bias,
-                                       'stddev': stddev,
-                                       'seed': self.seed}
-            return conv_out
 
-    def norm(self,
-             in_layer,
-             depth_radius=4,
-             bias=1.00, 
-             alpha=0.001 / 9.0,
-             beta=007,
-             name=None):
-        self.norm_counter += 1
-        self.global_counter += 1
-        if name is None:
-            name = 'norm' + str(self.norm_counter)
-        self.architecture[name] = {'input': in_layer.name,
-                                   'type': 'lrnorm',
-                                   'depth_radius': depth_radius,
-                                   'bias': bias,
-                                   'alpha': alpha,
-                                   'beta': beta}
-        return tf.nn.lrn(in_layer, 
-                         depth_radius=depth_radius, 
-                         bias=bias, 
-                         alpha=alpha, 
-                         beta=beta,
-                         name=name)
+        kernel = tf.get_variable(initializer=self.initializer(init, stddev=stddev),
+                                 shape=[ksize, ksize, in_shape, out_shape],
```

Comment:
do you think there's an argument for allowing non-square kernels?   For example:
    if ksize is an int:
        ksize1 = ksize2
    else
        ksize1, ksize2 = ksize

   and then call 
        kernel = tf.get_variable(...., shape=[ksize1, ksize2, ...])

And similarly for pooling?